native/src/modules/endpoint/reducers/refreshCategory.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import type { CategoryRouteStateType, CityContentStateType, PathType } from '../../app/StateType'
+import type { PushCategoryActionType } from '../../app/StoreActionType'
+import { CategoryModel, LanguageModel } from '@integreat-app/integreat-api-client'
+import forEachTreeNode from '../../common/forEachTreeNode'
+import ErrorCodes from '../../error/ErrorCodes'
+
+const getAllAvailableLanguages = (
+  category: CategoryModel, city: string, language: string, cityLanguages: Array<LanguageModel>
+) => {
+  if (category.isRoot()) {
+    return new Map(cityLanguages.map(model => [model.code, `/${city}/${model.code}`]))
+  }
+  const allAvailableLanguages = new Map(category.availableLanguages)
+  allAvailableLanguages.set(language, category.path)
+  return allAvailableLanguages
+}
+
+function extractResultModelsAndChildren (root: ?CategoryModel, categoriesMap: CategoriesMapModel, depth: number): {|
```

Comment:
Why is root allowed to be null?

Reply:
I copied this code from the very similar `pushCategory`. There, `root` is also allowed to be null. I don't know when exactly this might happen, but just left it this way.